### PR TITLE
Update recommendation engine APIs along with resource property fix, add new APIs

### PR DESCRIPTION
**Pull request by michimune**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR: [2548](https://github.com/Azure/azure-rest-api-specs/pull/2548)
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please pull down the latest changes and run:
-    msbuild build.proj /t:build /p:Scope=SDKs\Monitor
    generate.cmd to generate the code
-    These steps should generate the .props and .txt files, you need to commit these files

File: src/SDKs/WebSites/WebSites.Tests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -18,5 +18,5 @@
 
 [assembly: Guid("4c6e3586-2ebd-4b86-b46e-6850c1f5226e")]
 
-[assembly: AssemblyVersion("2.0.0.0")]
-[assembly: AssemblyFileVersion("2.0.0.0")]
+[assembly: AssemblyVersion("2.0.0.1")]
```
Comment (dsgouda) [inline src/SDKs/WebSites/WebSites.Tests/Properties/AssemblyInfo.cs]: Currently, the released package version is `1.7.1-preview` and the assembly version here is `2.0.0.0` please update the `csproj` with the new version number `2.0.0-preview` and undo these changes in `AssemblyInfo.cs`.

Comment (michimune) [reply src/SDKs/WebSites/WebSites.Tests/Properties/AssemblyInfo.cs]: Made changes.

File: src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Data Migration Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Data Migration Resources.")]
 
-[assembly: AssemblyVersion("0.1.2")]
-[assembly: AssemblyFileVersion("0.1.2")]
+[assembly: AssemblyVersion("0.2.0")]
```
Comment (dsgouda) [inline src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs]: Again, you don't need to update the assembly version here unless its a major version change.

File: src/SDKs/WebSites/WebSites.Tests/WebSites.Tests.csproj [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <PackageId>WebSites.Tests</PackageId>
     <Description>Websites.Tests Class Library</Description>
     <AssemblyName>WebSites.Tests</AssemblyName>
-    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <VersionPrefix>2.0.0-preview</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/WebSites/WebSites.Tests/WebSites.Tests.csproj]: nit: this is not necessary since we don't publish the tests as such

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks like there are two separate projects being modified in this PR `Microsoft.DataMigration` and `Microsoft.Websites`. The PR linked in description has changes related to the spec for `Microsoft.Websites` only, what was the intention behind making changes to two different projects here?

**Comment by dsgouda** [comment]
@shahabhijeet FYI

File: src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (michimune) [reply src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs]: Ah it was merged when I did git pull from Azure/azure-sdk-for-net. Sorry for confusion.

**Comment by michimune** [comment]
Closing the PR as per discussion with Deepak
